File: src/learning_gem5/tommy/example_object.cc
#include "learning_gem5/tommy/example_object.hh"

#include <iostream>

namespace gem5
{

ExampleObject::ExampleObject(const ExampleObjectParams &params)
    : SimObject(params)
{
    std::cout << "test number : " << params.number << std::endl;
    std::cout << "Hello World! From a SimObject!" << std::endl;
}

} // namespace gem5

File: src/learning_gem5/tommy/ExampleObject.py
from m5.params import *
from m5.SimObject import SimObject


class ExampleObject(SimObject):
    type = "ExampleObject"
    cxx_header = "learning_gem5/tommy/example_object.hh"
    cxx_class = "gem5::ExampleObject"
    number = Param.Int(0, "test number")
